client/src/index.ts

```diff
@@ -68,23 +70,28 @@ const makeLocalReadWrite = (name: string) => {
 };
 
 const abstractCache =
-  <X, Y>({ key, f, read, write }: AbstractCacheParams<X, Y>): Unary<X, Y> =>
-  (x: X) => {
-    const keyResult = key(x);
+  <X extends JSONArr, Y>({
+    key,
+    f,
+    read,
+    write,
+  }: AbstractCacheParams<X, Y>): Func<X, Y> =>
+  (...x: X) => {
+    const keyResult = key(...x);
```

Comment:
Why not just give `x` instead of spreading and un inside `key`?

Reply:
I think it’s nice that the key function accepts exactly the same arguments as the hashed function. If it has runtime/memory implications we can change this here without it affecting how keys are stored in the cache, no problem